src/apps/ENGINE/timer.h

```diff
@@ -1,112 +1,102 @@
 #pragma once
 
-#include <Windows.h>
-
-#define USE_HIGH_FREQUENCY
-#define FILTER_SIZE 16
+#include <chrono>
 #include <cstdint>
 
 class TIMER
 {
+    using Clock = std::chrono::high_resolution_clock;
+    using Milliseconds = std::chrono::milliseconds;
+    using time_point = std::chrono::time_point<Clock>;         // Time Point used to get time deltas.
+    using Duration = std::chrono::duration<float, std::milli>; // Time difference between time points.
+
   public:
-    TIMER() : FixedDeltaValue(0), Previous_Time(0), fDeltaTime(0)
+    bool Ring = false;            // If true, COMPILER::ProcessFrame adds a runtime event.
+    bool FixedDelta = false;      // Is the time delta between frames constant
+    uint32_t FixedDeltaValue = 0; // Value of constant time delta between frames
+    uint32_t Delta_Time = 20; // The time difference between current and last frame in milliseconds, as an unsigned int.
+    float fDeltaTime = 0.0f;  // The time difference between current and last frame, as a float
+    uint32_t rDelta_Time = Delta_Time;
+    uint32_t fps = 0;       // Frames per second based on the last second.
+    uint32_t fps_count = 0; // Running total of frames rendered in the last second, which resets to 0 every second.
+    uint32_t fps_time = 0;  // A one-second timer which is used to update the FPS counter on the screen.
+    const Milliseconds kMaxDelta = Milliseconds{100}; // Maximum time delta per frame.
+    time_point Current, Previous; // Timepoints of when each iteration of the timer has started and finished.
+
+    TIMER()
     {
-#ifdef USE_HIGH_FREQUENCY
-        LARGE_INTEGER liFreq;
-        QueryPerformanceFrequency(&liFreq);
-        fSecondsPerTick = 1.0f / liFreq.QuadPart;
-        QueryPerformanceCounter(&liPrevTime);
-#else
-        Previous_Time = GetTickCount();
-#endif
-        Delta_Time = 20;
-        rDelta_Time = Delta_Time;
-        fps = 0;
+        Current = Previous = Clock::now();
+    }
+
+    void RefreshFPS()
+    {
+        // Modulo operator is not necessary as we have limited maximum frame delta, so fps_time above 2000 milliseconds
+        // are not possible.
+        fps_time -= (uint32_t)Milliseconds{1000}.count();
```

Comment:
Consider making the `1000` a static const (something like `FpsUpdateInterval`) so it's extra clear what the number represents.

If you'd then also make `kMaxDelta` static, you could add a `static_assert(kMaxDelta < FpsUpdateInterval, "Max delta time should be less than fps update interval");` to verify the no-modulus requirement. Which can be very useful should the numbers ever be tweaked.